File: LeetCode/April Challenge/5th Day/Day_5.cpp
class Solution {
public:
    int maxProfit(vector<int>& p) {
        vector<int> a(p), b(p);
        sort(a.begin(), a.end(), greater<int>()); 
        sort(b.begin(),b.end());
        
        if(a==p)//check reverse sorted
            return 0;
        
        if(b==p)// check sorted
            return (b[b.size()-1]-b[0]);

        int profit=0;
        for(int i=0;i<(p.size()-1);i++)
        {
            if(p[i+1]>p[i])
                profit=profit+p[i+1]-p[i];
        }
        return profit;
    }
};

File: LeetCode/April Challenge/5th Day/Day_5.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        n = len(prices)
        ans = 0
        i = 0
        while i < (n - 1):
            while (i < (n - 1)) and prices[i + 1] <= prices[i]:
                i += 1
            if i == (n - 1):
                break
            buy = i
            i += 1
            
            while (i < n) and (prices[i] >= prices[i - 1]):
                i += 1
            sell = i - 1
            ans += prices[sell] - prices[buy]
        return ans
